aten/src/ATen/native/RNN.h

```diff
@@ -19,5 +19,34 @@ DECLARE_DISPATCH(rnn_packed_fn, gru_packed_cudnn_stub);
 DECLARE_DISPATCH(rnn_packed_fn, rnn_tanh_packed_cudnn_stub);
 DECLARE_DISPATCH(rnn_packed_fn, rnn_relu_packed_cudnn_stub);
 
-}} // namespace at::native
+void check_device(const Tensor& input, const TensorList& params, const TensorList& hiddens) {
+  auto input_device = input.device();
+  bool input_device_is_cuda = input_device.is_cuda();
+
+  auto check_tensors = [&](const std::string& name, const Tensor& t) {
+    if (!t.defined()) return;
+    auto t_device = t.device();
+    bool t_device_is_cuda = t_device.is_cuda();
```

Comment:
As I suggested, in my previous review, pretty much all of those could be simply `t.device() == input.device()` (you can cache `input.device()` outside of the loop).

Reply:
Thanks! I will make changes accordingly